session/session.go

```diff
@@ -2237,6 +2237,10 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 			s.currentPlan = nil
 		}
 	}
+	if variable.EnableResourceControl.Load() && domain.GetDomain(s).RunawayManager() != nil {
+		_, planDigest := executor.GetPlanDigest(sessVars.StmtCtx)
```

Comment:
This is because in buildExecutor, the plan changes.
We should move this after `e, err := a.buildExecutor()` in adapter.go